include/zephyr/bluetooth/ead.h

```diff
@@ -0,0 +1,147 @@
+ *       data with the advertising type and the length of those two.
+ *
+ * @param[in]  session_key Key of 16 bytes used for the encryption.
+ * @param[in]  iv Initialisation Vector used to generate the `nonce`.
+ * @param[in]  encrypted_payload Encrypted Advertising Data received. This
+ *             should only contain the advertising data from the received
+ *             advertising structure, not the length nor the type.
+ * @param[in]  encrypted_payload_size Size of the received advertising data in
+ *             bytes. Should be equal to the length field of the received
+ *             advertising structure, minus the size of the type (1 byte).
+ * @param[out] payload Decrypted advertising payload.
+ *
+ * @retval 0 Data have been correctly decrypted and authenticated.
+ * @retval -EIO Error occurred during the decryption or the authentication.
+ * @retval -EINVAL One of the argument is a NULL pointer or @p encrypted_payload_size is less than
+ *                 BT_EAD_RANDOMIZER_SIZE + BT_EAD_MIC_SIZE.
+ */
+int bt_ead_decrypt(const uint8_t *session_key, const uint8_t *iv, const uint8_t *encrypted_payload,
+		   size_t encrypted_payload_size, uint8_t *payload);
+
+/**
+ * @brief Helper function to get the size (in bytes) of the encrypted
+ * advertising data for a given set of `bt_data` structure.
+ *
+ * @param[in] ad_structs Array of `bt_data` structure.
+ * @param[in] num_elts Number of `bt_data` structure in @p ad_structs.
+ *
+ * @retval size_t Total size in bytes of the advertising structure from the
+ *                `bt_data` structure set.
+ */
+size_t bt_ead_get_encrypted_data_size(const struct bt_data *ad_structs, size_t num_elts);
+
+/**
+ * @brief Helper function to get the size (in bytes) of the advertising
+ * structure inside the AD Data of an encrypted advertising structure.
+ *
+ * @param[in] payload Advertising structure of type `BT_DATA_ENCRYPTED_AD_DATA`.
+ *
+ * @retval size_t Size in bytes of the advertising structure inside the AD data
+ *                of the @p payload advertising structure. If @p payload is not
+ *                an encrypted advertising structure, the function return 0.
+ */
+size_t bt_ead_get_decrypted_data_size(const struct bt_data *payload);
+
+/**
+ * @brief Helper function to convert `bt_data` structure into an advertising
+ * structure in an array. The data are formatted according to the Bluetooth Core
+ * Specification v. 5.4, vol. 3, part C, 11.
+ *
+ * @param[in]  input Single `bt_data` structure to read from.
+ * @param[out] output_ad_structure Buffer large enough to store the advertising
+ *             structure in @p input. The size of it must be at least the size
+ *             of the `AD data + 2` (for the type and the length).
+ *
+ * @retval size_t Number of bytes written in @p output_ad_structure.
+ */
+size_t bt_data_to_ad_structure(const struct bt_data *input, uint8_t *output_ad_structure);
+
+#ifdef CONFIG_BT_TESTING
```

Comment:
```suggestion
#if defined(CONFIG_BT_TESTING)
```

Not sure we should add test_specific code in the public APIs. It isn't uncommon for test application to access internal header files IIRC, so it could perhaps just be put there